src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java

```diff
@@ -50,24 +52,58 @@
 
 public class SubstrateDispatcher {
 
-    private static volatile boolean run = true;
+    private enum Step {
+        COMPILE(),
+        LINK(COMPILE),
+        PACKAGE(LINK),
+        INSTALL(PACKAGE),
+        RUN(INSTALL);
 
-    public static void main(String[] args) throws Exception {
+        private final Step dep;
 
-        String classPath = requireArg("imagecp","Use -Dimagecp=/path/to/classes");
-        String graalVM   = requireArg( "graalvm","Use -Dgraalvm=/path/to/graalvm");
+        Step() {
+            this.dep = null;
+        }
 
-        String mainClass = requireArg( "mainclass", "Use -Dmainclass=main.class.name" );
-        String appId     = Optional.ofNullable(System.getProperty("appId")).orElse("com.gluonhq.anonymousApp");
-        String appName   = Optional.ofNullable(System.getProperty("appname")).orElse("anonymousApp");
-        String targetProfile = System.getProperty("targetProfile");
+        Step(Step dep) {
+            this.dep = dep;
+        }
+
+        public boolean requires(Step step) {
+            return this == step || (dep != null && dep.requires(step));
+        }
+    }
+
+    private static volatile boolean compiling = true;
 
-        Triplet targetTriplet = targetProfile != null? new Triplet(Constants.Profile.valueOf(targetProfile.toUpperCase()))
-                :Triplet.fromCurrentOS();
+    public static void main(String[] args) throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'main(String)' has an NPath complexity of 480](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=5152426)